### PR TITLE
Add a prototype example docker-compose environment.

### LINKED ISSUE #238: Add example docker-compose setup to repo?
While https://www.modem7.com/books/docker-backup/page/backup-docker-using-borgmatic is well-written, I am personally the kind of person who works a *lot* better with a little example than I am with instructions. Would there be interest in adding a small minimally-functional example setup to the repo? I'm happy to send a pull request if so.

If not, and if I made my own repo with an example, would you be willing to link to it?

### PR DESCRIPTION
**Pull request by zorbathut**
This is very unfinished; it's basically what I put together from reading documentation. But man, having this would have made my setup a lot faster!

Suggestions and improvements welcome. I think having this is better than not having it, but there are absolutely ways to make it even better.

Closes #238 .

### CODE REVIEW

File: example/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version: "2.4"
+services:
+  borgmatic:
+    container_name: borgmatic
+
+    image: modem7/borgmatic-docker
```
Comment (grantbevis) [inline example/docker-compose.yml:6]: Should this be the docker-borgmatic ghcr.io container uri?

**Review by grantbevis — CHANGES_REQUESTED** [review]

File: example/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        # Borgmatic and crontab config files will go here.
+        - ./borgmatic.d:/etc/borgmatic.d/
+
+        # If you're using a remote repository, you'll need an SSH key.
+        # Uncomment this if you want to use root's ssh key.
+        #- ~/.ssh:/root/.ssh
+
+        # If you want a separate SSH key just for borg, uncomment this instead, then `ssh-keygen -f ssh/id_rsa`.
+        #- ./ssh:/root/.ssh
+
```
Comment (MarkJonas) [inline example/docker-compose.yml:34]: What about the Borgmatic state files?

`./.borgmatic:/root/.borgmatic`

Comment (zorbathut) [reply example/docker-compose.yml:34]: The [README](https://github.com/borgmatic-collective/docker-borgmatic/tree/master/base) says "this is now redundant and has been deprecated, please remove this from your configs", so I did. I checked my running install and there is a `checks` directory with a single 64-character-hex-hash subdirectory with a single 64-character-hex-hash-named empty file; I'm not sure what's going on there.

If you say this is important I'll put it back, but if so, the README should also be updated.

File: example/docker-compose.yml (hunk 1 above)
Comment (zorbathut) [reply example/docker-compose.yml]: Didn't even realize that existed :V I based this off [the link](https://www.modem7.com/books/docker-backup/page/backup-docker-using-borgmatic) at the bottom of [the README](https://github.com/borgmatic-collective/docker-borgmatic/blob/master/README.md) - it actually took a bit of work to find the ghcr.io container, even knowing it existed.

I've updated this!

File: example/docker-compose.yml (hunk 2 above)
Comment (MarkJonas) [reply example/docker-compose.yml:34]: I missed the [README](https://github.com/borgmatic-collective/docker-borgmatic/tree/master/base) update. You are right. It is not needed anymore.

File: example/docker-compose.yml (hunk 1 above)
Comment (zorbathut) [reply example/docker-compose.yml]: My local install, changed to this image, still works fine with a manual backup, I'll tell you tomorrow if the cron backup still works (though I have no reason to believe it won't)

Comment (zorbathut) [reply example/docker-compose.yml]: Seems to work, though it's a lot spammier and harder to read the log :V So it goes though.

**Comment by zorbathut** [comment]
Don't know if anyone is out there, but I'd like to get this in!

**Comment by grantbevis** [comment]
What benefit does this bring over the documentation we already have? There's a example docker-compose already and the borgmatic documentation is top notch.

**Comment by zorbathut** [comment]
Honestly, I didn't even see it when I was reading documentation, I think I'd ended up clicking on the modem7 link 

However, it also isn't really a single self-contained example, it's kind of awkwardly interlaced with the base image definition. And the docker-compose that exists doesn't really provide much in the way of explanation for someone who doesn't already know how it works; a bunch of unfilled-out env variables doesn't help without sensible defaults.

(it's also out of date because it still includes .borgmatic)

I think an important goal for examples is an example that can be followed along and filled out, and that's what I'm going for here. Some people (like myself) work best from examples.

**Pull request closed without merging**